plans/run.pp

```diff
@@ -65,7 +67,11 @@
     true    => { 'noop' => true, },
     default => {},
   }
-  $args = $arg_env + $arg_noop + { '_catch_errors' => true }
+  $arg_tags = $tags ? {
+    true    => { 'tags' => $tags },
+    default => {},
+  }
```

Comment:
This would only happen if `$tags == true` (boolean true, not any truthy value, this is not what you want).

```suggestion
  $arg_tags = if $tags {
    { 'tags' => $tags }
  } else {
    {}
  }
```

Reply:
on interesting

Reply:
to stay consistent with the above logic, I keep the selector but validate the `Undef` datatype now